### ISSUE TITLE
Java agent recipe fails with 'Property proxy_hostname must be one of: String!  You passed nil.'

### ISSUE THREAD

**Issue by MitchellJThomas** [Tripwire/chef-tripwire_agent#16]
In a Vagrantfile using Chef Solo providing the chef JSON, if I don't explicitly specify a string value for proxy_hostname, I get the error message:   ` Chef::Exceptions::ValidationFailed Property proxy_hostname must be one of: String!  You passed nil.`

```
==> proxy: Installing Cookbook Gems:
==> proxy: Compiling Cookbooks...
==> proxy: 
==> proxy: ================================================================================
==> proxy: Recipe Compile Error in /var/chef/cache/cookbooks/tripwire_agent/recipes/java_agent.rb
==> proxy: ================================================================================
==> proxy: 
==> proxy: Chef::Exceptions::ValidationFailed
==> proxy: ----------------------------------
==> proxy: Property proxy_hostname must be one of: String!  You passed nil.
==> proxy: 
==> proxy: Cookbook Trace:
==> proxy: ---------------
==> proxy:   /var/chef/cache/cookbooks/tripwire_agent/recipes/java_agent.rb:15:in `block in from_file'
==> proxy:   /var/chef/cache/cookbooks/tripwire_agent/recipes/java_agent.rb:6:in `from_file'
==> proxy: 
==> proxy: Relevant File Content:
==> proxy: ----------------------
==> proxy: /var/chef/cache/cookbooks/tripwire_agent/recipes/java_agent.rb:
==> proxy: 
==> proxy:   8:    console node['tripwire_agent']['java']['console']
==> proxy:   9:    services_password node['tripwire_agent']['java']['services_password']
==> proxy:  10:    console_port node['tripwire_agent']['java']['console_port']
==> proxy:  11:    install_directory node['tripwire_agent']['java']['install_directory']
==> proxy:  12:    install_rtm node['tripwire_agent']['install_rtm']
==> proxy:  13:    rtm_port node['tripwire_agent']['rtm_port']
==> proxy:  14:    proxy_agent node['tripwire_agent']['java']['proxy_agent']
==> proxy:  15>>   proxy_hostname node['tripwire_agent']['proxy_hostname']
==> proxy:  16:    proxy_port node['tripwire_agent']['proxy_port']
==> proxy:  17:    fips node['tripwire_agent']['java']['fips']
==> proxy:  18:    integration_port node['tripwire_agent']['java']['integration_port']
==> proxy:  19:    start_service node['tripwire_agent']['start_service']
==> proxy:  20:    tags node['tripwire_agent']['tags']
==> proxy:  21:  end
==> proxy:  22:  
==> proxy:  23:  # Might be better to leave starting the service out of the custom resource
==> proxy:  24:  # Customers might want to manage service control their own way instead of
==> proxy: 
==> proxy: System Info:
==> proxy: ------------
==> proxy: chef_version=13.7.16
==> proxy: platform=centos
==> proxy: platform_version=7.4.1708
==> proxy: ruby=ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]
==> proxy: program_name=chef-solo worker: ppid=19773;start=18:35:25;
==> proxy: executable=/opt/chef/bin/chef-solo
```

Vagrantfile example (To use this file install the vagrant-berkshelf and vagrant-vbguest to be installed)
```
Vagrant.require_version '>= 1.8.6'

Vagrant.configure(2) do |config|
  config.berkshelf.enabled = true

  config.vm.box = "centos/7"
  config.omnibus.chef_version = :latest

  config.vm.define 'proxy-test' do |proxy_agent|
    proxy_agent.vm.provider :virtualbox do |v, override|
      v.memory = 512
      v.cpus = 1
    end

    proxy_agent.vm.provision :chef_solo do |chef|
      chef.json = {
        :tripwire_agent => {
          :installer => "http://<agent download server hostname here>/te_agent.bin",
          :java => {
            :console => 'te-console-hostname',
            :services_password => 'te-console-service-password',
            :proxy_agent => true,
          }
        }
      }
      chef.run_list = [
        'recipe[tripwire_agent::java_agent]',
      ]
    end
  end
end
```
Berksfile
```
source "https://supermarket.chef.io"
cookbook 'tripwire_agent'
```

**Comment by bespokebob** [comment]
Looks like #10 fixed this, and then #11 broke it again?
I wonder if the ghost of @effective-spoons would visit us long enough to explain what happened there...

**Comment by MitchellJThomas** [comment]
I should have been clearer about my intention in the Vagrantfile example above.  That file should create a java agent proxy, not an agent for which to communicate with a proxy.   To state the hopefully obvious, the proxy_hostname resource is used by non-proxy agents to connect with a java proxy agent.   

Here is my proposed fix (pretty simple stuff)  #17

**Comment by bespokebob** [comment]
Closed by #17
